src/items/items.js

```diff
@@ -170,8 +170,45 @@ class Item {
    * @type {string|null}
    */
   get previousState () {
-    const state = this.rawItem.getLastState();
-    return state == null ? null : state.toString();
+    return this.previousRawState == null ? null : this.previousRawState.toString();
+  }
+
+  /**
+   * Numeric representation of Item previous state, or `null` if state is not numeric or not available.
+   * @type {number|null}
+   */
+  get previousNumericState () {
+    const previousState = this.previousRawState == null ? null : this.previousRawState.toString();
+    if (previousState == null) return null;
+    const numericState = parseFloat(previousState);
+    return isNaN(numericState) ? null : numericState;
+  }
+
+  /**
+   * Previous item state as {@link Quantity} or `null` if state is not Quantity-compatible, Quantity would be unit-less (without unit) or not available.
+   * @type {Quantity|null}
+   */
+  get previousQuantityState () {
+    const previousState = this.previousRawState == null ? null : this.previousRawState.toString();
+    if (previousState == null) return null;
```

Comment:
Why doing an additional null check here?

Reply:
previousRawState can be null

Reply:
Correct, I would simplify the code by first checking for null and early returning instead of checking twice for null.
The toString method shouldn’t return null for a non-null type.